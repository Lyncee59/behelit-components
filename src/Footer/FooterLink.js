import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Link from '../Links/Link'

const Wrapper = styled(Link).attrs({
  size: '1rem',
  weight: 500
})`
  position: relative;
  white-space: nowrap;
`

const FooterLink = ({ children, ...rest }) => <Wrapper {...rest}>{children}</Wrapper>

FooterLink.propTypes = {
  children: PropTypes.node.isRequired
}

export default FooterLink
