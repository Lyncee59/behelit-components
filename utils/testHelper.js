import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/Theme'

export const mountWithTheme = (tree) => mount(<ThemeProvider>{tree}</ThemeProvider>)