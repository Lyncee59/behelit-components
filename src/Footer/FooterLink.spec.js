import React from 'react'
import { mountWithTheme } from '../../utils/testHelper'

import FooterLink from './FooterLink'

describe('FooterElement', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<FooterLink>FooterLink</FooterLink>)

    expect(tree.find(FooterLink)).toMatchSnapshot()
  })
})
