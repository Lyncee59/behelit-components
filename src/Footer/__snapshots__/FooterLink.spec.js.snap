// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterElement should render correctly 1`] = `
.c1 {
  font-family: 'Roboto',sans-serif;
  font-weight: 500;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #990000;
  opacity: 1;
  cursor: pointer;
}

.c1:hover {
  color: #800000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  position: relative;
  white-space: nowrap;
}

<FooterLink>
  <Styled(Link)>
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "size": "1rem",
              "weight": 500,
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  position: relative;
  white-space: nowrap;
",
            ],
          },
          "displayName": "Styled(Link)",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      forwardedRef={null}
    >
      <Link
        capitalize={false}
        className="c0"
        italic={false}
        opacity={1}
        size="1rem"
        uppercase={false}
        weight={500}
      >
        <styled.a
          capitalize={false}
          className="c0"
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={500}
        >
          <StyledComponent
            capitalize={false}
            className="c0"
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  font-family: ",
                    [Function],
                    ";
  font-weight: ",
                    [Function],
                    ";
  font-size: ",
                    [Function],
                    ";
  text-transform: ",
                    [Function],
                    ";
  font-style: ",
                    [Function],
                    ";
  color: ",
                    [Function],
                    ";
  opacity: ",
                    [Function],
                    ";
  cursor: pointer;

  &:hover {
    color: ",
                    [Function],
                    ";
    text-decoration: underline;
  }
",
                  ],
                },
                "displayName": "styled.a",
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "a",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={500}
          >
            <a
              className="c0 c1"
              opacity={1}
              size="1rem"
            >
              FooterLink
            </a>
          </StyledComponent>
        </styled.a>
      </Link>
    </StyledComponent>
  </Styled(Link)>
</FooterLink>
`;
